library/modules/EventManager.cpp

```diff
@@ -173,28 +173,88 @@ static void manageInteractionEvent(color_ostream& out);
 
 typedef void (*eventManager_t)(color_ostream&);
 
-static const eventManager_t eventManager[] = {
-        manageTickEvent,
-        manageJobInitiatedEvent,
-        manageJobStartedEvent,
-        manageJobCompletedEvent,
-        manageNewUnitActiveEvent,
-        manageUnitDeathEvent,
-        manageItemCreationEvent,
-        manageBuildingEvent,
-        manageBuildingCreatedEvent,
-        manageBuildingDestroyedEvent,
-        manageConstructionEvent,
-        manageConstructionAddedEvent,
-        manageConstructionRemovedEvent,
-        manageSyndromeEvent,
-        manageInvasionEvent,
-        manageInventoryChangeEvent,
-        manageReportEvent,
-        manageUnitAttackEvent,
-        manageUnloadEvent,
-        manageInteractionEvent,
-};
+// integrate new events into this function, and no longer worry about syncing the enum list with the `eventManager` array
+eventManager_t getManager(EventType::EventType t) {
+    switch (t) {
+        case EventType::TICK:
+            return manageTickEvent;
+        case EventType::JOB_INITIATED:
+            return manageJobInitiatedEvent;
+        case EventType::JOB_STARTED:
+            return manageJobStartedEvent;
+        case EventType::JOB_COMPLETED:
+            return manageJobCompletedEvent;
+        case EventType::UNIT_NEW_ACTIVE:
+            return manageUnitNewActiveEvent;
+        case EventType::UNIT_DEATH:
+            return manageUnitDeathEvent;
+        case EventType::ITEM_CREATED:
+            return manageItemCreationEvent;
+        case EventType::BUILDING_DESTROYED:
+            return manageBuildingDestroyedEvent;
+        case EventType::BUILDING_CREATED:
+            return manageBuildingCreatedEvent;
+        case EventType::BUILDING:
+            return manageBuildingEvent;
+        case EventType::CONSTRUCTION_REMOVED:
+            return manageConstructionRemovedEvent;
+        case EventType::CONSTRUCTION_ADDED:
+            return manageConstructionAddedEvent;
+        case EventType::CONSTRUCTION:
+            return manageConstructionEvent;
+        case EventType::SYNDROME:
+            return manageSyndromeEvent;
+        case EventType::INVASION:
+            return manageInvasionEvent;
+        case EventType::INVENTORY_CHANGE:
+            return manageInventoryChangeEvent;
+        case EventType::REPORT:
+            return manageReportEvent;
+        case EventType::UNIT_ATTACK:
+            return manageUnitAttackEvent;
+        case EventType::UNLOAD:
+            return manageUnloadEvent;
+        case EventType::INTERACTION:
+            return manageInteractionEvent;
+        case EventType::EVENT_MAX:
+            return nullptr;
```

Comment:
Maybe add a comment here that you are deliberately not including a default case so that the compiler will warn/error on missing enums

Reply:
sounds like a plan

Reply:
done

Reply:
> Also in eventful.cpp

Do you think it might be more beneficial to actually include in eventful? (a default case)

Reply:
now that the construction of the array is "safe", we *could* put a default case in, but why would we ever want an event to *not* be handled by eventful? is there such a thing as a C++-only event?

Reply:
if we always want events to be exported to lua, then it would make more sense to just do what eventful does in the event manager over having a plugin provide that